File: app/src/main/java/ds/photosight/ui/ListFragmentAbstract.java
package ds.photosight.ui;

import java.util.HashMap;
import java.util.Map;

import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.support.v4.app.ListFragment;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import com.flurry.android.FlurryAgent;
import ds.photosight.Constants;
import ds.photosight.R;
import ds.photosight.utils.L;

public abstract class ListFragmentAbstract extends ListFragment implements Constants {

	protected int mPositionChecked = 0;
	protected int mPositionShown = -1;


	private boolean isActive() {
		return getUserVisibleHint();
	}


	@Override
	public void onViewCreated(final View view, final Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		L.v("list on view created " + getListType());

		mPositionChecked = getRoot().getListSelection(getListType());
		if (Build.VERSION.SDK_INT > 8)
			getListView().setOverScrollMode(View.OVER_SCROLL_NEVER);

		ArrayAdapter<String> adapter = initList();
		setListAdapter(adapter);

		getListView().setChoiceMode(ListView.CHOICE_MODE_SINGLE);
		getListView().setSelection(mPositionChecked);
		getListView().setItemChecked(mPositionChecked, true);
	}


	@Override
	public void onListItemClick(ListView l, View v, int position, long id) {

		showDetails(position);

		flurry(((TextView) v.findViewById(android.R.id.text1)).getText().toString());

		super.onListItemClick(l, v, position, id);
	}


	private void flurry(String text) {
		Map<String, String> args = new HashMap<String, String>();
		args.put("Category", text);
		args.put("Tab", String.valueOf(getListType()));
		FlurryAgent.logEvent(LIST_CLICKED, args);
		Log.d(args.get("Tab"), args.get("Category"));
	}


	@Override
	public void onSaveInstanceState(Bundle b) {
		/*b.putInt("item", mPositionShown);
		if (!App.getInstance().isPortrait())
			b.putBoolean("fromLandscape", true);*/
		super.onSaveInstanceState(b);
	}


	private MainActivity getRoot() {
		return (MainActivity) getActivity();
	}


	void showDetails(int index) {
		if (!isActive())
			return;

		mPositionChecked = index;
		getListView().setItemChecked(index, true);

		getRoot().setCurrPage(0);
		getRoot().selectItem(index);

		mPositionShown = index;
	}


	protected abstract int getListType();


	protected abstract ListAdapter initList();


	//
	// **********************************************************************************************************************************************
	// ListAdapter
	// **********************************************************************************************************************************************
	//
	public class ListAdapter extends ArrayAdapter<String> {

		public ListAdapter(Context context, int textViewResourceId, String[] objects) {
			super(context, textViewResourceId, objects);
		}


		@Override
		public View getView(int position, View convertView, ViewGroup parent) {
			View v = super.getView(position, convertView, parent);

			if (position == mPositionChecked) {
				v.setBackgroundResource(R.drawable.selector_color_solid);
			} else {
				v.setBackgroundResource(R.drawable.list_selector);
			}

			return v;
		}

	}
}

File: app/src/main/java/ds/photosight/ui/ListFragmentAbstract.kt
package ds.photosight.ui

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.fragment.app.ListFragment
import ds.photosight.Constants
import ds.photosight.utils.L

public abstract class ListFragmentAbstract : ListFragment(), Constants {

    protected var mPositionChecked: Int = 0
    protected var mPositionShown: Int = -1


    private fun isActive(): Boolean {
        return userVisibleHint
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        L.v("list on view created " + listType)

        mPositionChecked = getRoot().getListSelection(listType)
        listView.overScrollMode = View.OVER_SCROLL_NEVER

        val adapter = initList()
        listAdapter = adapter

        listView.choiceMode = AbsListView.CHOICE_MODE_SINGLE
        listView.setSelection(mPositionChecked)
        listView.setItemChecked(mPositionChecked, true)
        listView.setSelector(android.R.color.transparent)
    }


    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        super.onListItemClick(l, v, position, id)
        showDetails(position)
    }

    private fun getRoot(): MainActivity {
        return activity as MainActivity
    }


    private fun showDetails(index: Int) {
        if (!isActive())
            return

        mPositionChecked = index
        listView.setItemChecked(index, true)

        getRoot().currPage = 0
        getRoot().selectItem(index)

        mPositionShown = index
    }


    protected abstract val listType: Int


    protected abstract fun initList(): ListAdapter?


    //
    // **********************************************************************************************************************************************
    // ListAdapter
    // **********************************************************************************************************************************************
    //
    class ListAdapter(context: Context, textViewResourceId: Int, objects: Array<String>) : ArrayAdapter<String>(context, textViewResourceId, objects) {

    }
}
